python/pypto/jit/specializer.py

```
@@ -1510,6 +1525,22 @@ def _specialize_function(self, ctx: SpecializeContext) -> list[str]:
+        # (e.g. ``tids: pl.Array[N, pl.TASK_ID]``) — render the annotation with
+        # closure constants folded so the generated source stays parseable.
+        array_param_anns: dict[str, str] = {}
+        for arg in func_def.args.args:
+            if arg.arg == "self" or arg.annotation is None:
+                continue
+            if arg.arg in tensor_params or arg.arg in scalar_dtype_strs:
+                continue
+            try:
+                ann_obj = eval(ast.unparse(arg.annotation), dict(ctx.py_globals))  # noqa: S307
+            except Exception:  # noqa: BLE001 — best effort; unrecognized stays bare
+                continue
```

Ruff (0.15.15)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)